core/domain/question_services.py

```diff
@@ -217,3 +226,141 @@ def update_question(
+    """
+    question_html_data = question.question_data['content']['html']
+    question_summary = question_domain.QuestionSummary(
+        question.id, creator_id, question_html_data)
+
+    return question_summary
+
+
+def save_question_summary(question_summary):
+    """Save a question summary domain object as a QuestionSummaryModel
+    entity in the datastore.
+
+    Args:
+        question_summary: The question summary object to be saved in the
+            datastore.
+    """
+    question_summary_model = question_models.QuestionSummaryModel(
+        id=question_summary.id,
+        creator_id=question_summary.creator_id,
+        question_model_last_updated=question_summary.last_updated,
+        question_model_created_on=question_summary.created_on,
+        question_html_data=question_summary.question_html_data
+    )
+
+    question_summary_model.put()
+
+
+def get_question_summaries_by_creator_id(creator_id):
+    """Gets question summaries of questions created by the user.
+
+    Args:
+        creator_id: str. The user ID of the creator.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    return question_models.QuestionSummaryModel.get_by_creator_id(creator_id)
+
+
+def get_question_summary_by_question_id(question_id, strict=False):
+    """Gets question summaries of questions created by the user.
+
+    Args:
+        question_id: str. The user ID of the question.
+        strict: bool. Whether to fail noisily if no question summary for the
+            given question id exists in the datastore.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    return question_models.QuestionSummaryModel.get(question_id, strict=strict)
+
+
+def get_summaries_of_linked_skills(question_id):
+    """Gets linked skill IDs for given question.
+
+    Args:
+        question_id: str. The question ID for the given question.
+
+    Returns:
+        QuestionSkillLinkModel|None. The QuestionSkillModel for the given
+            question or None if there exists no skill linked for question.
+    """
+    linked_skill_summaries = []
+    question_skill_links = question_models.QuestionSkillLinkModel.get(
+        question_id, strict=False)
+
+    if question_skill_links is None:
+        return None
+    for question_skill_link in question_skill_links:
+        linked_skill_summaries = (
+            skill_models.SkillSummaryModel.get(
+                question_skill_link.skill_id, strict=False))
+
+    return linked_skill_summaries
+
+
+def check_can_edit_question(user_id, question_id):
+    """Checks if the user can edit the given question or not
+
+    Args:
+        user_id: str. The user ID of the user.
+        question_id: str. The question ID of the question.
+
+    Returns:
+        bool. Represents if the user can edit the question or not.
+    """
+    question_summary = get_question_summary_by_question_id(
```

Comment:
We should not be using question_summary, which is a transient regeneratable model, to compute rights. There should be a QuestionRights object instead, similar to explorations and collections.

Reply:
Yes, changed